src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -200,17 +207,23 @@ private String getLink(String id, String query) {
     return null;
   }
 
-  private String getPreferredSite(String userId) {
+  private String[] getPreference(String userId) {
+    String[] result = new String[2];
```

Comment:
This changes the API. Before, it would return "" when a preference was not found, but now it returns null. This would cause a NPE in Aaron's code, which assumes it would only be "", not null. Can you initialize to a literal array of two empty strings?